pkg/rtc/room_egress.go

```diff
@@ -0,0 +1,58 @@
+package rtc
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/livekit/livekit-server/pkg/rtc/types"
+	"github.com/livekit/protocol/livekit"
+)
+
+type EgressLauncher interface {
+	StartEgress(context.Context, *livekit.StartEgressRequest) (*livekit.EgressInfo, error)
+}
+
+func (r *Room) startTrackEgress(track types.MediaTrack) {
+	if r.egressLauncher == nil {
+		r.Logger.Errorw("egress launcher not found", nil)
+		return
+	}
+
+	output := &livekit.TrackEgressRequest_File{
+		File: &livekit.DirectFileOutput{},
+	}
+
+	if prefix := r.protoRoom.Egress.Tracks.FilePrefix; prefix != "" {
+		if strings.HasSuffix(prefix, "/") {
+			output.File.Filepath = prefix
+		} else {
+			output.File.Filepath = fmt.Sprintf("%s_%s", prefix, track.ID())
```

Comment:
@frostbyte73 Will you consider to use UNIX timestamp as well as? Having timestamp will be helpful for post processing task.